ch_integration_tests/tests/integration.rs

```diff
@@ -718,4 +718,96 @@ mod tests {
 
         assert!(r.is_ok());
     }
+
+    #[test]
+    fn test_bridge_networking() {
+        use std::fs;
+        use std::io::Write;
+        use std::process::Command;
+        cleanup_libvirt_state();
+
+        //configure host IP  since we cant do it from XML for bridge network.
+        Command::new("sh")
+                .arg("-c")
+                .arg("sudo ip addr add 192.168.1.1/24 dev mybr0 ; sudo ip addr add 192.168.2.1/24 dev mybr0")
+                .spawn()
+                .expect("Failed to set IP address.");
+
+        let mut guest_name: [String; 2] = ["guest1".to_string(), "guest2".to_string()];
+        let mut guest_ip: [String; 2] = ["ip1".to_string(), "ip2".to_string()];
+
+        let mut libvirtd = spawn_libvirtd().unwrap();
+        thread::sleep(std::time::Duration::new(5, 0));
+
+        for i in 0..2 {
+            let mut disk = UbuntuDiskConfig::new(FOCAL_IMAGE_NAME.to_owned());
+            let guest = Guest::new(&mut disk, KernelType::RustFw);
+            let domain_path = guest.create_domain(VcpuConfig::default(), DEFAULT_RAM_SIZE);
+
+            let mut contents =
+                fs::read_to_string(&domain_path).expect("Something went wrong reading the file");
+            contents = contents.replace("ethernet", "bridge");
+            contents = contents.replace("</source>", "");
+            contents = contents.replace("source><ip ", "source bridge=\"mybr0\" ");
+
+            let mut f = std::fs::File::create(&domain_path).unwrap();
+            f.write_all(contents.as_bytes()).unwrap();
+
+            let r = std::panic::catch_unwind(|| {
+                spawn_virsh(&["create", domain_path.to_str().unwrap()])
+                    .unwrap()
+                    .wait()
+                    .unwrap();
+                guest.wait_vm_boot(None).unwrap();
+            });
+
+            let guestname = &guest.vm_name;
+            guest_name[i] = guestname.to_string();
+            assert!(r.is_ok());
+
+            guest_ip[i] = guest
+                .ssh_command(
+                    "ip route get 8.8.8.8 | sed -n '/src/{s/.*src *\\([^ ]*\\).*/\\1/p;q}'",
```

Comment:
Ah that's a nice way to get an ip without mask of an interface in use!

Reply:
Thanks :)